hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/LinkedListStreamSerializer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.serialization.impl;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.StreamSerializer;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+/**
+ * The {@link java.util.LinkedList} serializer
+ */
+public class LinkedListStreamSerializer
+        implements StreamSerializer<LinkedList> {
+
+    @Override
+    public void write(ObjectDataOutput out, LinkedList object)
+            throws IOException {
+        out.writeBoolean(object != null);
+        if (object != null) {
+            int size = object.size();
+            out.writeInt(size);
+            Iterator iterator = object.iterator();
```

Comment:
Probably best to make a check when the size == 0, in this case no iterator needs to be created.